finalfrontier-utils/src/bin/ff-deps.rs

```diff
@@ -139,3 +153,36 @@ where
 
     (input_builder.into(), output_builder.into())
 }
+
+struct SentenceIter<P, R> {
+    inner: Reader<R>,
+    projectivizer: Option<P>,
+}
+
+impl<P, R> SentenceIter<P, R> {
+    fn new(read: R, projectivizer: Option<P>) -> Self where R: BufRead {
+        SentenceIter {
+            inner: Reader::new(read),
+            projectivizer
+        }
+    }
+}
+
+impl<P, R> Iterator for SentenceIter<P, R> where P: Projectivize, R: BufRead {
+    type Item = Sentence;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if let Some(proj) = &self.projectivizer {
+
+            self.inner
```

Comment:
This can perhaps be simplified to something like:

```rust
let mut sentence = self.inner.read_sentence().or_exit("Cannot read sentence", 1)?;
if let Some(proj) = &self.projectivizer {
  proj.projectivize(&mut sentence);
}
```

Reply:
Didn't know the `?` operator worked like this on options, neat!